show-version-diff/run.js

```diff
@@ -0,0 +1,133 @@
+
+// Work out old versions to apply the change to
+const lookup = {
+  "gateway-oss": "ce",
+  deck: "deck",
+  enterprise: "ee",
+  "getting-started-guide": "gsg",
+  "kubernetes-ingress-controller": "kic",
+  konnect: "konnect",
+  mesh: "mesh",
+  gateway: "gateway",
+};
+
+if (!argv.source) {
+  throw new Error("Missing --source option");
+}
+
+// Check if the file exists
+if (!fs.existsSync(sourceFileFull)) {
+  throw new Error(sourceFile + " does not exist");
+}
+
+// Extract product from the source file (it's the segment after `app`)
+const parts = sourceFile.split("/");
+const appIndex = parts.findIndex((n) => n == "app");
+if (appIndex === -1) {
+  throw new Error(
+    `Invalid --source provided: [${source}]. Must start with [app]`
+  );
+}
+
+const product = parts[appIndex + 1];
+const shortProduct = lookup[product];
+
+if (!lookup[product]) {
+  console.log(
+    `Invalid --product: [${product}]. Valid values:\n${Object.keys(lookup)
+      .map((n) => `* ${n}`)
+      .join("\n")}`
+  );
+}
+
+const path = product + "_*";
+let files = fg.sync(`../app/_data/docs_nav_${path}.yml`);
+files = files
+  .map((f) => {
+    if (f.includes("docs_nav_contributing")) {
+      return;
+    }
+    const item = { path: f };
+    const info = f.replace("../app/_data/docs_nav_", "").replace(/\.yml$/, "");
+
+    // Special case Konnect Platform
+    if (info === "konnect_platform") {
+      item.type = "konnect-platform";
+      item.version = "";
+    } else if (info === "konnect") {
+      item.type = "konnect";
+      item.version = "";
+    } else {
+      const x = info.split("_");
+      item.type = lookup[x[0]];
+      item.version = semver.coerce(x[1]).raw;
+      item.original_version = x[1];
+    }
+    return item;
+  })
+  .filter((n) => n)
+  // Remove anything older than the minimum version
+  .filter((n) => !ignore.includes(n.original_version))
+  // Sort oldest to newest
+  .sort((a, b) => (semver.gte(a.version, b.version) ? 1 : -1));
+
+const newestVersion = files[files.length - 1].original_version;
+for (let i in files) {
+  const current = files[i];
+  const next = files[++i];
+  //console.log(next);
+  if (!next) {
+    continue;
+  }
+  const currentPath = sourceFileFull.replace(
+    newestVersion,
+    current.original_version
+  );
+  const nextPath = sourceFileFull.replace(newestVersion, next.original_version);
+  try {
+    const output = execa("git", [
+      "--no-pager",
+      "diff",
+      "--ignore-all-space",
+      "--color=always",
+      "--no-index",
+      "--unified=1",
+      currentPath,
+      nextPath,
+    ]);
+    console.log(
+      `No differences between ${currentPath} and ${nextPath}\n-----------------\n`
+    );
+  } catch (e) {
+    // Diff always returns exit code 1, so we have to catch
+    console.log(
+      `##########################################\n${currentPath}\n${nextPath}\n##########################################`
+    );
+    if (e.stdout.length > 0) {
```

Comment:
I keep getting the following error when I try to run this:

![Screen Shot 2022-03-21 at 10 53 30 AM](https://user-images.githubusercontent.com/54370747/159335450-9919fd20-3c8f-4c16-9124-fca34d6dd238.png)

I tried this with a bunch of different files, including the decK one that your provided in the example. Not sure what the difference is here...

Reply:
It uses `git diff` under the hood, so it should be portable. Might be one to hop on a call and poke around to see what's going on

Reply:
Messed around with this a bunch, still no luck (though a regular `git diff` works fine). I think line 127 isn't reading from stdout for me, for whatever reason. We can do a short call tomorrow/later this week to figure this one out.